custom_connectors/custom_auth/trackvia_connector.rb

```diff
@@ -0,0 +1,769 @@
+          toggle_field: call(:get_toggle_field,
+                             type: field['type'],
+                             name: field['name'],
+                             required: !field['required'])
+        }
+      end
+    end,
+
+    get_fields_sample_output: lambda do |input|
+      view_id = input[:view_id]
+      get("/openapi/views/#{view_id}?start=0&max=1")
+        .dig('data', 0)
+    end
+  },
+  object_definitions: {
+    app: {
+      fields: lambda do
+        [
+          { name: 'name' },
+          { name: 'id', type: :integer }
+        ]
+      end
+    },
+    view: {
+      fields: lambda do
+        [
+          { name: 'id', type: :integer },
+          { name: 'name' },
+          { name: 'applicationName' },
+          { name: 'default', type: :boolean }
+        ]
+      end
+    },
+    record: {
+      fields: lambda { |_connection, config_fields|
+        call(:get_fields, view_id: config_fields['view_id'])
+      }
+    },
+    response_record: {
+      fields: lambda { |_connection, config_fields|
+        call(:get_output_fields, view_id: config_fields['view_id'])
+      }
+    },
+    hook_body: {
+      fields: lambda { |_connection, config_fields|
+        call(:get_output_fields, view_id: config_fields['view_id'])
+      }
+    },
+    request: {
+      fields: lambda do |_connection, config_fields|
+        [
+          {
+            name: 'data',
+            type: :array,
+            of: :object,
+            properties: [
+              call(:get_fields, view_id: config_fields['view_id'])
+            ]
+          }
+        ]
+      end
+    },
+    user: {
+      fields: lambda do
+        [
+          { name: 'Status' },
+          { name: 'Time Zone' },
+          { name: 'Email' },
+          { name: 'Record ID' },
+          { name: 'Updated', type: :date_time },
+          { name: 'First Name' },
+          { name: 'id', type: :integer },
+          { name: 'Last Name' },
+          { name: 'Created', type: :date_time }
+        ]
+      end
+    },
+    column: {
+      fields: lambda do
+        [
+          { name: 'name' },
+          { name: 'type' },
+          { name: 'required', type: :boolean },
+          { name: 'unique', type: :boolean },
+          { name: 'canRead', type: :boolean },
+          { name: 'canUpdate', type: :boolean },
+          { name: 'canCreate', type: :boolean },
+          { name: 'displayOrder', type: :integer },
+          { name: 'relationshipSize', type: :integer },
+          { name: 'propertyName' }
+        ]
+      end
+    }
+  },
+  pick_lists: {
+    apps: ->(_connection) { get('/openapi/apps').pluck('name', 'name') },
+    views: lambda do |_connection, app_name:|
+      views = get('/openapi/views')
+              .after_error_response(/.*/) do |_code, body, _header, message|
+                error("#{message} : #{body}")
+              end
+      views.select { |view| view['applicationName'] == app_name }
```

Comment:
Ruby follows chaining methods e.g.
`views.select { |view| view['applicationName'] == app_name }&.pluck('name', 'id')`
& - is the safe operator when block returns` nil`

Does this block has issue?
```
      get('/openapi/views')
        .after_error_response(/.*/) do |_code, body, _header, message|
          error("#{message} : #{body}")
        end&.select { |view| view['applicationName'] == app_name }
        &.pluck('name', 'id')
```

Reply:
That does indeed work. My rubocop is trying to use the Ruby 2.2 parser and flagged that syntax as an Error. I've fixed the code and my linter.